File: accfin/app/api/routes/vendor_suggestions.py
"""Vendor name autocomplete for parsing confirmation."""

from __future__ import annotations

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db_session
from app.core.dependencies import require_permission
from app.schemas.auth import TokenData
from app.schemas.vendor_suggestion import VendorSuggestionResponse
from app.services.vendor_suggestions import get_vendor_suggestions

router = APIRouter(prefix="/vendor-suggestions", tags=["Vendor suggestions"])


@router.get("", response_model=list[VendorSuggestionResponse])
async def list_vendor_suggestions(
    search: str = Query(..., min_length=2),
    limit: int = Query(10, ge=1, le=20),
    _user: TokenData = Depends(require_permission("cases:read")),
    session: AsyncSession = Depends(get_db_session),
) -> list[VendorSuggestionResponse]:
    return await get_vendor_suggestions(session, search, limit=limit)

File: finance-ui/src/lib/api/vendor-suggestions.ts
import { apiFetch } from './client';

export type VendorSuggestion = {
  name: string;
  source: 'counterparty' | 'case_history';
  counterparty_type: string | null;
  email: string | null;
};

export function fetchVendorSuggestions(search: string, limit = 10) {
  const q = new URLSearchParams({ search, limit: String(limit) });
  return apiFetch<VendorSuggestion[]>(`/vendor-suggestions?${q}`);
}
